src/java/org/apache/cassandra/concurrent/InfiniteLoopExecutor.java

```diff
@@ -53,12 +53,17 @@ public enum InternalState { SHUTTING_DOWN_NOW, TERMINATED }
     public enum SimulatorSafe { SAFE, UNSAFE }
 
     /**
-     * Does this loop always block on some external work provision that is going to be simulator-controlled, or does
-     * it loop periodically? If the latter, it may prevent simulation making progress between phases, and should be
-     * marked as a DAEMON process.
+     * Simulator Tag specifies the nature of the created thread:
+     * 
+     *   * JOB threads are short-lived, and simulation will wait for them to terminate before completing an enclosing Work unit
```

Comment:
JOB threads are short-lived and simulation treats them as _sub tasks_ of the task that creates them, so that the strictly ordered property of the simulator ensures the thread terminates before the next task of its parent is scheduled.